cordapp-upgrades/README.md

```diff
@@ -0,0 +1,158 @@
+versions.
+
+The CorDapp is divided into contracts and workflows. These are versioned separately.
+
+# Current versions
+
+The CorDapp contract and workflow versions are described below.
+
+## Contracts
+
+### Version 1
+
+The initial contract version.
+
+### Version 2
+
+This second version adds a new field to the obligation state that allows the borrower to default on the obligation. A
+command is added to the contract to facilitate this. The new field has to be nullable in order to ensure backwards
+compatibility between nodes with the old contract version and nodes with the new contract version while a rolling
+upgrade is carried out.
+
+## Workflows
+
+### Version 1
+The initial version of the CorDapp. This is the Obligation CorDapp using the version of FinalityFlow that was present in
+Corda 3.
+
+Compiled against version 1 of the contracts CorDapp.
+
+### Version 2
+This version of the CorDapp upgrades to use the new version of the FinalityFlow present in Corda 4. To do this such that
+ the app remains compatible with Version 1, the following must be done:
+ - The flows using the new FinalityFlow must be versioned
+ - The flows must detect if the counterparty is using the old version, and revert to the old API accordingly
+   
+Compiled against version 1 of the contracts CorDapp.
+
+### Version 4
+This version adds the flows required to default on an Obligation.
+
+Compiled against version 2 of the contracts CorDapp.
+
+# Upgrade scenarios
+
+This section described the upgrade scenarios that can be explored using the workflows and contracts in this CorDapp.
+Note that all commands should be run from the root of the CorDapp directory structure (i.e. the location of this Readme).
+
+## Scenario 1: Finality Flow Upgrade
+
+This scenario covers upgrading an app to run on V4 by upgrading its usage of FinalityFlow to the new API defined in
+Corda 4. It utilises the flow versioning API to determine what version of the flow is running on the counterparty and
+respond accordingly. This process can be generalised to any flow upgrade procedure that may be required.
+
+ 1. Deploy the nodes in their initial configuration by running `./gradlew deployNodes`, then `build/nodes/runnodes`.
+ This creates a network of 3 nodes, all running version 1 of the contract. PartyA and PartyB are using the original
+ version of the workflows, while PartyC is using version 2.
+ 
+ 2. Check that all the nodes can transact with each other. This can be done manually from the shell, or by running
+ `./gradlew issueBetweenNodes`. This will issue an obligation between each pair of nodes in the network.
+ 
+ 3. Now shut down the nodes and upgrade all nodes to version 2 of the workflows CorDapp. The upgrade part of this can be
+ carried out by running `scripts/upgradeNodes.sh workflows v2-new-finality-flow PartyA PartyB`.
+ This simply copies the workflow jars (and any additional configuration) to the cordapps/ directory on the nodes specified.
```

Comment:
mention any specific configuration that has changed between V1 and V2.

Reply:
The cordapp configuration has been removed, which is what this was referring to.